rmc/utils/generic.py

```diff
@@ -702,17 +714,33 @@ def import_de_novo_variants(overwrite: bool, n_partitions: int = 5000) -> None:
         Will also help determine number of partitions in final Table.
     :return: None; writes HT to resource path.
     """
-    # TODO: Re-import this HT with samples collected into lists rather than sets
     fu_ht_path = f"{TEMP_PATH_WITH_FAST_DEL}/fu_dn.ht"
     kaplanis_ht_path = f"{TEMP_PATH_WITH_FAST_DEL}/kaplanis_dn.ht"
     if not file_exists(fu_ht_path) or overwrite:
         import_fu_data(overwrite=overwrite)
-        fu_ht = hl.read_table(fu_ht_path, _n_partitions=n_partitions)
     if not file_exists(kaplanis_ht_path) or overwrite:
         import_kaplanis_data(overwrite=overwrite)
-        kap_ht = hl.read_table(kaplanis_ht_path, _n_partitions=n_partitions)
 
+    fu_ht = hl.read_table(fu_ht_path, _n_partitions=n_partitions)
+    kap_ht = hl.read_table(kaplanis_ht_path, _n_partitions=n_partitions)
     ht = kap_ht.join(fu_ht, how="outer")
+
     # Union sample types (DD, ASD, control)
-    ht = ht.transmute(sample_set=ht.case_control.union(ht.role))
+    ht = ht.transmute(
+        role_str=hl.str(",").join(ht.role),
+        case_control_str=hl.str(",").join(ht.case_control),
+    )
+    ht = ht.transmute(
+        sample_set=hl.case()
+        .when(
+            hl.is_defined(ht.case_control_str) & hl.is_defined(ht.role_str),
+            ht.case_control_str + ht.role_str,
+        )
+        .when(
+            hl.is_defined(ht.case_control_str) & hl.is_missing(ht.role_str),
+            ht.case_control_str,
+        )
+        .default(ht.role_str)
+    )
+    ht = ht.transmute(sample_set=ht.sample_set.split(","))
```

Comment:
We should just be able to do this, right? Unless I'm missing something
```
ht1 = ht.transmute(
    sample_set = hl.case()
    .when(hl.is_defined(ht.case_control) & hl.is_defined(ht.role), ht.case_control.extend(ht.role))
    .when(hl.is_defined(ht.case_control), ht.case_control)
    .when(hl.is_defined(ht.role), ht.role)
    .or_missing()
)
```

Reply:
chatted on slack; adding this because the extend worked for Lily